src/App.js

```diff
@@ -1,16 +1,44 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+
+let totalLikes = 0;
+
+  const updateLikes = (chatData) => {
+    let likes = 0;
+    for (const chat of chatData) {
+      if (chat.liked) {
+        likes += 1;
+      } 
+    }
+    totalLikes = likes;
+  };
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+  const updateChatData = (updatedChat) => {
+    const entries = chatData.map((newChat) => {
+      if (newChat.id === updatedChat.id) {
+        return updatedChat;
+      } else {
+        return newChat;
+      }
+    });
+    setChatData(entries);
+    updateLikes(entries);
+  };
```

Comment:
This is OK, but I think it better fits the React paradigm of lifting state to actually toggle the liked setting here, and only pass up the id of entry that needs to have its `liked` member toggled rather than the entire updated chat entry.